riot/riot.py

```diff
@@ -556,19 +571,54 @@ def run(
                 assert command is not None
                 if cmdargs is not None:
                     command = command.format(cmdargs=(" ".join(cmdargs))).strip()
-                env_str = " ".join(f"{k}={v}" for k, v in env.items())
+                env_str = "\n".join(f"{k}={v}" for k, v in env.items())
                 logger.info(
-                    "Running command '%s' with environment '%s'.", command, env_str
+                    "Running command '%s' in venv '%s' with environment:\n%s.",
+                    command,
+                    venv_path,
+                    env_str,
                 )
+                # Copy over *nspkg.pth files so that we can handle namespace packages.
+                # This is strictly required by Python 2.7 and <3.3 (see PEP 420).
+                ns_files = []
                 try:
-                    # Pipe the command output directly to `out` since we
-                    # don't need to store it.
+                    sitepkgs = inst.site_packages_path
+                    assert sitepkgs is not None, inst
+                    for path in (_ for _ in inst.site_packages_list if _ != sitepkgs):
```

Comment:
```suggestion
                    if inst.python_version < (3, 3):
                        for path in (_ for _ in inst.site_packages_list if _ != sitepkgs):
```

?

Reply:
That might work too, but from tests it doesn't look like it hurts doing the same thing for all Python versions so I'd keep the current code complexity. WDYT?

Reply:
yeah, figured if we can lock the workaround to older versions only then would reduce chances of bugs, at least for newer versions.

I don't have a really strong preference here, just a thought.